### PR TITLE
Add script to communicate with networking hardware database

### LINKED ISSUE #46: Get database, back-end server, and front-end UI all communicating (with server run locally)

### PR DESCRIPTION
**Pull request by julianstirling**
This creates a way to update the JSON file from the noCoDB endpoint.

Closes #46

### CODE REVIEW

**Comment by julianstirling** [comment]
@jmwright this one should be ready.

This is only failing on the linter because of a new pylint checker. I think the new pylint checker will improve the code so I have not disabled it. I have opened #94 separately to fix this issue.

**Comment by jmwright** [comment]
@julianstirling Is fixing #94 going to end up causing merge conflicts with #92 ?

**Comment by julianstirling** [comment]
If you want we can review and merge #92 as it is. Then do #94 afterwards? I can do a review today if you want to pull it out of draft?

**Comment by jmwright** [comment]
>  I can do a review today if you want to pull it out of draft?

I've only done the assembly for the RPi shelf as an example case to get the rendering working. I could do a separate PR that does assembly for the other shelf types. Thoughts?

**Comment by julianstirling** [comment]
I think progress with no serious regressions is a good criteria for merging. No need to finish everything in one MR, just complicates working in parallel. Happy to merge #92 as a proof of principle for the renders.

**Comment by jmwright** [comment]
@julianstirling This is showing merge conflicts now. Do you want me to review it as-is, or do you want to resolve the conflicts first?

**Comment by julianstirling** [comment]
@jmwright I have rebased it so it should be free of merge conflicts now

File: nimble_build_system/utils/nimble_devices_updater.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    """
+    Main script to turn a CSV file into a JSON file. It does not do pretty formatting,
+    it is a JSON file with no newlines.
+    """
+
+    args = parse_args()
+
+    if args.local:
+        if args.url:
+            raise RuntimeError("Cannot set an API path and a local file. "
```
Comment (jmwright) [inline nimble_build_system/utils/nimble_devices_updater.py:68]: My code preview is showing separate strings in spots like this that seem like they should have a single string. Is this correct?

File: setup.py [hunk 2]
```diff
@@ -17,10 +17,10 @@
         'cadscript>=0.5.2',
         'exsource-tools',
         'cq-cli @ git+https://github.com/CadQuery/cq-cli.git',
-        'gitbuilding>=0.0.6',
+        'gitbuilding==0.15.0a4',
```
Comment (jmwright) [inline setup.py:20]: Isn't this the GitBuilding version that caused the problem the other day?

**Review by jmwright — APPROVED** [review]
@julianstirling Just a couple comments. Otherwise it looks good, thanks.

File: setup.py (hunk 2 above)
Comment (julianstirling) [reply setup.py:20]: No it was alpha 2 with the bug, but we need the version 15 alphas for the features added during the NLNet project. I missed what version had been set when reviewing #92 . The `>=0.0.6` was the setting for exsource that GitBuilding had written as `==0.0.6`. With this configuration it should all work.

**Pull request merged**